aws/resource_aws_servicecatalog_portfolio.go

```diff
@@ -159,14 +148,14 @@ func resourceAwsServiceCatalogPortfolioUpdate(d *schema.ResourceData, meta inter
 	}
 
 	if d.HasChange("tags") {
-		currentTags, requiredTags := d.GetChange("tags")
-		log.Printf("[DEBUG] Current Tags: %#v", currentTags)
-		log.Printf("[DEBUG] Required Tags: %#v", requiredTags)
-
-		tagsToAdd, tagsToRemove := tagUpdates(requiredTags.(map[string]interface{}), currentTags.(map[string]interface{}))
-		log.Printf("[DEBUG] Tags To Add: %#v", tagsToAdd)
-		log.Printf("[DEBUG] Tags To Remove: %#v", tagsToRemove)
-		input.AddTags = tagsToAdd
+		o, n := d.GetChange("tags")
+
+		input.AddTags = keyvaluetags.New(n).IgnoreAws().ServicecatalogTags()
+
+		var tagsToRemove []*string
+		for k := range keyvaluetags.New(o).IgnoreAws().Map() {
+			tagsToRemove = append(tagsToRemove, aws.String(k))
+		}
 		input.RemoveTags = tagsToRemove
```

Comment:
I believe this can be simplified via the below. 👍 

```go
input.RemoveTags = aws.StringSlice(keyvaluetags.New(o).IgnoreAws().Keys())
```

Reply:
done

Reply:
--- PASS: TestAccAWSServiceCatalogPortfolio_Basic (35.47s)
--- PASS: TestAccAWSServiceCatalogPortfolio_Disappears (28.91s)
--- PASS: TestAccAWSServiceCatalogPortfolio_Tags (84.47s)